test/c_api_tests.cpp

```diff
@@ -283,6 +285,52 @@ TEST(CAPITests, GetOutputCAPI) {
   generator->GenerateNextToken();
 }
 
+TEST(CAPITests, SetTerminate) {
+#if TEST_PHI2
+
+  auto GeneratorSetTerminateCall = [](OgaGenerator* generator) {
+    generator->SetTerminate();
+  };
+
+  auto GenerateOutput = [](OgaGenerator* generator, std::unique_ptr<OgaTokenizerStream> tokenizer_stream) {
+    try {
+      while (!generator->IsDone()) {
+        generator->ComputeLogits();
+        generator->GenerateNextToken();
+      }
+    }
+    catch (const std::exception& e) {
+      std::cout << "Session Terminated: " << e.what() << std::endl;
```

Comment:
Can we detect session termination vs another error here?

Reply:
Added, thanks.